File: abc/abc123/abc123_b.cs
// https://atcoder.jp/contests/abc123/submissions/8438152

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var input = new int[5].Select(_ => int.Parse(Console.ReadLine())).ToArray();

            var index = 0;
            var value = input[0] % 10;
            for (int i = 0; i < 5; i++)
            {
                if (0 < input[i] % 10 && input[i] % 10 < value)
                {
                    value = input[i] % 10;
                    index = i;
                }
            }

            var result = 0;
            for (int i = 0; i < 5; i++)
            {
                if (i != index)
                {
                    result += (int)Math.Ceiling(input[i] / 10.0) * 10;
                }
                else
                {
                    result += input[i];
                }
            }
            Console.WriteLine(result);
        }
    }
}

File: abc/abc123/abc123_b.fs
// https://atcoder.jp/contests/abc123/submissions/12198189

let x = Array.zeroCreate<int> 5

for i in [ 0..4 ] do
    x.[i] <- stdin.ReadLine() |> int

let mutable index = 0
let mutable value = x.[0] % 10

for i in [ 0..4 ] do
    if 0 < x.[i] % 10 && x.[i] % 10 < value then
        value <- x.[i] % 10
        index <- i

let mutable result = 0

for i in [ 0..4 ] do
    if i <> index then result <- result + 10 * int (ceil ((float x.[i]) / 10.))
    else result <- result + x.[i]

result |> stdout.WriteLine
